Dockerfile

```diff
@@ -52,16 +186,62 @@ USER 65532:65532
 
 ENTRYPOINT ["/manager"]
 
-FROM gcr.io/distroless/static:nonroot as populator
-WORKDIR /
-COPY --from=populator-bin /workspace/bin/populator .
-USER 65532:65532
 
-ENTRYPOINT ["/populator"]
 
-FROM gcr.io/distroless/static:nonroot as ori-volume
+# Since we're leveraging apt to pull in dependencies, we use `gcr.io/distroless/base` because it includes glibc.
+FROM gcr.io/distroless/base-debian11 as distroless-base
+
+# The distroless amd64 image has a target triplet of x86_64
+FROM distroless-base AS distroless-amd64
+ENV LIB_DIR_PREFIX x86_64
+ENV LIB_DIR_PREFIX_MINUS x86-64
+
+# The distroless arm64 image has a target triplet of aarch64
+FROM distroless-base AS distroless-arm64
+ENV LIB_DIR_PREFIX aarch64
+ENV LIB_DIR_PREFIX_MINUS aarch64
+
+
+FROM busybox:1.35.0-uclibc as busybox
+FROM distroless-$TARGETARCH  as ori-volume
 WORKDIR /
-COPY --from=ori-volume-bin /workspace/bin/ori-volume .
+COPY --from=busybox /bin/sh /bin/sh
+COPY --from=busybox /bin/mkdir /bin/mkdir
+COPY --from=ori-volume-builder /lib/x86_64-linux-gnu/librados.so.2 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/librbd.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libc.so.6 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libfmt.so.9 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libstdc++.so.6 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libgcc_s.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libssl.so.3 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libcryptsetup.so.12 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libcrypto.so.3 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libresolv.so.2 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libboost_thread.so.1.74.0 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libboost_iostreams.so.1.74.0 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libblkid.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libudev.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libibverbs.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/librdmacm.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libm.so.6 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libuuid.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libdevmapper.so.1.02.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libargon2.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libjson-c.so.5 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libz.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libbz2.so.1.0 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/liblzma.so.5 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libzstd.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libnl-route-3.so.200 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libnl-3.so.200 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libselinux.so.1 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libpthread.so.0 \
+/lib/${LIB_DIR_PREFIX}-linux-gnu/libpcre2-8.so.0 /lib/${LIB_DIR_PREFIX}-linux-gnu
+COPY --from=ori-volume-builder /lib64/ld-linux-${LIB_DIR_PREFIX_MINUS}.so.2 /lib64/
```

Comment:
I think it might make sense to have a 'validation' image in between validating that all shared libraries are present.
For reference, we do the same in `onmetal-csi-driver`:
https://github.com/onmetal/onmetal-csi-driver/blob/fddb2662974d1db48f8efdb3ff878c155431060a/Dockerfile#L109